power_grid_model_c/power_grid_model/include/power_grid_model/job_dispatch.hpp

```diff
@@ -14,23 +14,20 @@
 
 namespace power_grid_model {
 
-template <class MainModel, class... ComponentType> class JobDispatch {
-  private:
-    using SequenceIdxView = std::array<std::span<Idx2D const>, main_core::utils::n_types<ComponentType...>>;
-
+class JobDispatch {
   public:
     static constexpr Idx ignore_output{-1};
     static constexpr Idx sequential{-1};
 
-    template <typename Calculate>
-        requires std::invocable<std::remove_cvref_t<Calculate>, MainModel&, MutableDataset const&, Idx>
-    static BatchParameter batch_calculation_(MainModel& model, CalculationInfo& calculation_info,
-                                             Calculate&& calculation_fn, MutableDataset const& result_data,
-                                             ConstDataset const& update_data, Idx threading = sequential) {
-        // if the update dataset is empty without any component
-        // execute one power flow in the current instance, no batch calculation is needed
+    // TODO(figueroa1395): remove calculation_fn dependency
```

Comment:
Are these TODOs for a later stage or should they be addressed here?

Reply:
but the PR would become too big to do them now